### PR TITLE
nvcodec: update example text to reflect HEVC

### PR DESCRIPTION
**Pull request by bradh**
The text says MPEG-1, but the capability being checked is HEVC.

### CODE REVIEW

**Comment by bradh** [comment]
@devjeonghwan If you have an opportunity, can you take a look at this small patch?

**Comment by saudet** [comment]
@devjeonghwan Does this look OK to you?

**Comment by devjeonghwan** [comment]
@bradh
I missed something wrong!

The original sample intent was to "always succeed". So I wrote sample code that check "MPEG-1" codec support. (because all of nvidia graphic card is support MPEG-1)

[Video Decoder Capabilities Table](https://docs.nvidia.com/video-technologies/video-codec-sdk/nvdec-video-decoder-api-prog-guide/index.html#video-decoder-capabilities__table_o3x_fms_3lb)

Here is my suggest.
```java
// Query decoder capability 'MPEG-1' codec
{
    CUVIDDECODECAPS decodeCaps = new CUVIDDECODECAPS();
    decodeCaps.eCodecType(cudaVideoCodec_MPEG1);
    decodeCaps.eChromaFormat(cudaVideoChromaFormat_420);
    decodeCaps.nBitDepthMinus8(0);

    checkCudaApiCall("cuvidGetDecoderCaps", cuvidGetDecoderCaps(decodeCaps));

    System.out.printf("Decoder Capability MPEG-1 Codec\t : %s \r\n", (decodeCaps.bIsSupported() != 0));
}
```

File: nvcodec/README.md [hunk 1]
```diff
@@ -112,7 +112,7 @@ public class SampleEncodeDecode {
             System.out.printf("Encoder Max Supported Version\t : %d \r\n", version.get());
         }
 
-        // Query decoder capability 'MPEG-1' codec
+        // Query decoder capability for HEVC (H.265) codec
         {
             CUVIDDECODECAPS decodeCaps = new CUVIDDECODECAPS();
             decodeCaps.eCodecType(cudaVideoCodec_HEVC);
```
Comment (devjeonghwan) [inline nvcodec/README.md:118]: Change `cudaVideoCodec_HEVC` to `cudaVideoCodec_MPEG1`

File: nvcodec/samples/simple/SampleEncodeDecode.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public static void main(String[] args) {
             System.out.printf("Encoder Max Supported Version\t : %d \r\n", version.get());
         }
 
-        // Query decoder capability 'MPEG-1' codec
+        // Query decoder capability for HEVC (H.265) codec
         {
             CUVIDDECODECAPS decodeCaps = new CUVIDDECODECAPS();
             decodeCaps.eCodecType(cudaVideoCodec_HEVC);
```
Comment (devjeonghwan) [inline nvcodec/samples/simple/SampleEncodeDecode.java:43]: Change `cudaVideoCodec_HEVC` to `cudaVideoCodec_MPEG1`

**Review by devjeonghwan — CHANGES_REQUESTED** [review]

**Comment by bradh** [comment]
> @bradh I missed something wrong!
> 
> The original sample intent was to "always succeed". So I wrote sample code that check "MPEG-1" codec support. (because all of nvidia graphic card is support MPEG-1)

OK. However having that code in the README.md might make the nvcodec capabilities look fairly limited. Would it be OK to have three (e.g. H.264/AVC - supported on all cards, H.265/HEVC - supported on more recent cards, and AV1 - supported on only the most recent GPUs)? The code should still run and always succeed on the first, but shows a wider range of capabilities.

Unrelated to that, is any cleanup on the context (`cuContext`) required (for the general case)?

Brad

**Comment by devjeonghwan** [comment]
@bradh 

> OK. However having that code in the README.md might make the nvcodec capabilities look fairly limited. Would it be OK to have three (e.g. H.264/AVC - supported on all cards, H.265/HEVC - supported on more recent cards, and AV1 - supported on only the most recent GPUs)? The code should still run and always succeed on the first, but shows a wider range of capabilities.

Yeah, Sounds good! We can use`H.264` instead of `MPEG-1`.

> Unrelated to that, is any cleanup on the context (cuContext) required (for the general case)?

Yes, `Simple Sample` didn't care about resource cleanup. If necessary, we need to change it as below.

```java
int targetGpu = 0; // If you use NVIDIA GPU not '0', changing it.

CUctx_st cuContext = new CUctx_st();

checkCudaApiCall("cuInit", cuInit(0));
checkCudaApiCall("cuCtxCreate", cuCtxCreate(cuContext, 0, targetGpu));
try {
    // Check encoder max supported version
    {
        try (IntPointer version = new IntPointer(1)) {
            checkEncodeApiCall("NvEncodeAPIGetMaxSupportedVersion", NvEncodeAPIGetMaxSupportedVersion(version));

            System.out.printf("Encoder Max Supported Version\t : %d \r\n", version.get());
        }
    }

    // Query decoder capability 'MPEG-1' codec
    {
        try (CUVIDDECODECAPS decodeCaps = new CUVIDDECODECAPS()) {
            decodeCaps.eCodecType(cudaVideoCodec_MPEG1);
            decodeCaps.eChromaFormat(cudaVideoChromaFormat_420);
            decodeCaps.nBitDepthMinus8(0);

            checkCudaApiCall("cuvidGetDecoderCaps", cuvidGetDecoderCaps(decodeCaps));

            System.out.printf("Decoder Capability MPEG-1 Codec\t : %s \r\n", (decodeCaps.bIsSupported() != 0));
        }
    }
} finally {
    checkCudaApiCall("cuCtxDestroy", cuCtxDestroy(this.context));
}
```

(`nvcodec/samples/src/...` Samples contain resource cleanup.)

**Pull request closed without merging**